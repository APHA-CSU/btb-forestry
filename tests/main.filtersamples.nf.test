nextflow_process {

    name "Test Process filterSamples"
    script "main.nf"
    process "FILTER_SAMPLES"

    test("Should filter samples") {

        when {
            params {
                today = "01Jan25"
            }
            process {
                """
                input[0] = [ 'B6-16', 
                file("${projectDir}/tests/data/split_exp.csv"),
                '52532',
                'test',
                'today' ]
                input[1] = file("${projectDir}/tests/data/testoutlier.txt")
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
