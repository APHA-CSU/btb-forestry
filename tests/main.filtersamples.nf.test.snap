{
    "Should filter samples": {
        "content": [
            {
                "0": [
                    [
                        "B6-16",
                        "B6-16_01Jan25_samplelist.csv:md5,ca6c8ef4fdf182eacd2746c5fb52b347"
                    ]
                ],
                "1": [
                    "B6-16_01Jan25_highN.csv:md5,99a9ae0bc31f0e742d90ae44dff04410"
                ],
                "excludedSamples": [
                    "B6-16_01Jan25_highN.csv:md5,99a9ae0bc31f0e742d90ae44dff04410"
                ],
                "includedSamples": [
                    [
                        "B6-16",
                        "B6-16_01Jan25_samplelist.csv:md5,ca6c8ef4fdf182eacd2746c5fb52b347"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-08-22T13:54:26.628960885"
    }
}